sklearn/ensemble/_forest.py

```diff
@@ -118,6 +118,12 @@ def _get_n_samples_bootstrap(n_samples, max_samples):
         return max_samples
 
     if isinstance(max_samples, Real):
+        if (n_samples * max_samples) < 1:
```

Comment:
This will break the following:

```python
from sklearn.datasets import make_classification
from sklearn.ensemble import RandomForestClassifier

X, y = make_classification(n_samples=100)
forest = RandomForestClassifier(max_samples=0.007)

forest.fit(X, y)
```

In this case `max_samples` will end up to be exactly 1 because `round(n_samples * max_samples)` is 1.

Instead of checking `n_samples * max_samples < 1`, we can check the result directly:

```python
if isinstance(max_samples, Real):
    output = round(n_samples * max_samples)
    if result < 1:
        raise ValueError(...)
    return output
```